presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -395,26 +396,8 @@ private void processNewSources()
                 }
             }
 
-            for (int index = activeOperators.size() - 1; index >= 0; index--) {
-                if (activeOperators.get(index).isFinished()) {
-                    // close and remove this operator and all source operators
-                    List<Operator> finishedOperators = this.activeOperators.subList(0, index + 1);
-                    Throwable throwable = closeAndDestroyOperators(finishedOperators);
-                    finishedOperators.clear();
-                    if (throwable != null) {
-                        throwIfUnchecked(throwable);
-                        throw new RuntimeException(throwable);
-                    }
-                    // Finish the next operator, which is now the first operator.
-                    if (!activeOperators.isEmpty()) {
-                        Operator newRootOperator = activeOperators.get(0);
-                        newRootOperator.getOperatorContext().startIntervalTimer();
-                        newRootOperator.finish();
-                        newRootOperator.getOperatorContext().recordFinish();
-                    }
-                    break;
-                }
-            }
+            // Find the highest finished operator and destroy all operators below it.
```

Comment:
IMO this comment belongs more to the `for` loop inside `clearFinishedOperators` method